repository/src/main/java/org/jbpm/process/workitem/repository/storage/InMemoryRepositoryStorage.java

```diff
@@ -76,16 +77,29 @@ public void onUninstalled(RepoData service, String target) {
         // no op
 
     }
-    
+
     protected void enforceId(RepoData service) {
         if (service.getId() == null) {
             try {
-                service.setId(UUID.nameUUIDFromBytes(service.getName().getBytes("UTF-8")).toString());
-            } catch (UnsupportedEncodingException e) {                
+                service.setId(UUID.nameUUIDFromBytes(generateServiceId(service)).toString());
+            } catch (UnsupportedEncodingException e) {
             }
         }
     }
 
+    private byte[] generateServiceId(RepoData service) throws UnsupportedEncodingException {
+        if (service.getGav() != null) {
+            return (service.getGav().toString() + ":" + service.getName()).getBytes(StandardCharsets.UTF_8);
+        }
+
+        if (service.getMavenDependencies().size() > 0) {
```

Comment:
what are we doing here? We're naming the service after one of its dependencies?

Reply:
the service task name is repeatable and not unique. if you upload different GAV's service task but they have same name,them's id is same.